checker/src/lib.rs

```diff
@@ -1,12 +1,15 @@
 #![doc = include_str!("../README.md")]
+#![deny(clippy::all)]
 #![allow(
 	unreachable_code,
 	unused_variables,
 	unused_imports,
 	unused_mut,
 	dead_code,
 	irrefutable_let_patterns,
-	deprecated
+	deprecated,
+    // TODO: Remove when fixed
+	clippy::result_unit_err
```

Comment:
Added this one to by pass the `Result<_, ()>` linting as it needs more than a refactor I think